src/datasets/tasks/text_classification.py

```diff
@@ -23,30 +24,26 @@ def __get__(self, obj, objtype=None):
 
 @dataclass(frozen=True)
 class TextClassification(TaskTemplate):
-    task: ClassVar[str] = "text-classification"
+    # `task` is not a ClassVar since we want it to be part of the `asdict` output for JSON serialization
+    task: str = "text-classification"
     input_schema: ClassVar[Features] = Features({"text": Value("string")})
     # TODO(lewtun): Find a more elegant approach without descriptors.
     label_schema: ClassVar[Features] = FeaturesWithLazyClassLabel(Features({"labels": ClassLabel}))
-    labels: List[str]
     text_column: str = "text"
     label_column: str = "labels"
+    labels: Optional[Tuple[str]] = None
 
     def __post_init__(self):
-        assert len(self.labels) == len(set(self.labels)), "Labels must be unique"
-        # Cast labels to tuple to allow hashing
-        self.__dict__["labels"] = tuple(sorted(self.labels))
+        if self.labels:
+            assert len(self.labels) == len(set(self.labels)), "Labels must be unique"
+            # Cast labels to tuple to allow hashing
+            self.__dict__["labels"] = tuple(sorted(self.labels))
+            self.__dict__["label_schema"] = copy.deepcopy(self.label_schema)
```

Comment:
Should be safe to remove the `FeaturesWithLazyClassLabel` descriptor after this to reduce the code complexity (and then the deepcopy call can be replaced with `self.label_schema.copy()`).

Reply:
yes i agree, thanks for the suggestion! since we now create a new instance of `TextClassification` in `DatasetInfo.__post_init__` I think we should be safe from the side-effects you spotted earlier